### PR TITLE
Get image and machine info from apiserver in node e2e test

### PR DESCRIPTION
**Pull request by coufon**
This PR changes node e2e test to get image and machine information from API server instead of pass them from Jenkins test framework. The original format to pass image and machine info is naming the test node as "machine-image-uuid", which is hard to parse because "-" occurs a lot in both machine and image names.

Now we add two labels "image" and "machine" into performance data. The machine type has the format "cpu:1core,memory:3.6GB".

This PR is based on #32250.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32342)

### CODE REVIEW

**Comment by coufon** [comment]
@dchen1107  @Random-Liu

**Comment by dchen1107** [comment]
@k8s-bot test this issue: #30075

**Comment by coufon** [comment]
@k8s-bot test this issue: #31706

**Comment by dchen1107** [comment]
Please rebase your branch, so I can start the review.

**Comment by coufon** [comment]
@k8s-bot test this issue #31884

**Comment by coufon** [comment]
@k8s-bot test this issue #31884

**Comment by dchen1107** [comment]
LGTM

**Comment by coufon** [comment]
@k8s-bot  test this issue:  #32574

**Comment by coufon** [comment]
@k8s-bot test this issue: #32574

**Comment by dchen1107** [comment]
LGTM
